src/lib/compound.util.js

```diff
@@ -321,19 +322,20 @@ export default class CToken {
         return await wApiAction(txPromise, user, web3, ethToSendWithTransaction, onHash)
     }
 
-    repay = async (amount, onHash) => {
+    repay = async (repayAmount, onHash) => {
         const {web3, networkType, user} = userStore
-        const reapyAmount = fromUiDeciamlPointFormat(amount, this.tokenInfo.underlyingDecimals)
+        const reapyAmount = fromUiDeciamlPointFormat(repayAmount, this.tokenInfo.underlyingDecimals)
         let ethToSendWithTransaction
         let txPromise
         if(this.symbol === "ETH"){
             txPromise = CI.repayEth(web3, networkType, this.address)
             ethToSendWithTransaction = reapyAmount
-        }else {
-            const isMax = this.getMaximum(ActionEnum.repay) === amount
+        }else { 
+            const debt = this.getMaximum(ActionEnum.repay)
+            const repayAmountIs99Point99PrecentOfDebt = percentage(repayAmount, debt) >= 99.99
```

Comment:
seems to be wrong. need to convert back to non big number? or keep a copy of non big number.

Reply:
these values are parsed using toUiDecimalPointFormat
which means they are floats